### ISSUE TITLE
Use flow-maven-plugin in the Bakery to bundle things

### ISSUE THREAD

**Issue by SomeoneToIgnore** [vaadin/flow#3319]
@SomeoneToIgnore commented on [Wed Jan 10 2018](https://github.com/vaadin/flow-maven-plugin/issues/41)

We need the real application using the plugin bundling features so we can 
* determine what's the performance impact of bundling
* determine missing or oddly working features in the plugin

For now it is enough to have two fragments. One for just the login page needed resources, and one for everything else. It might be nice to know the difference in performance between having just one fragment (the default) compared to the split. 
  

---

@denis-anisimov commented on [Fri Jan 12 2018](https://github.com/vaadin/flow-maven-plugin/issues/41#issuecomment-357230908)

There are a number of issue with the plugin (#45, #46, #47) which blocks this task.
I stop to work on it for now.
It might be that the bakery project is not a good one for this specific task since it's Spring Boot based and it introduces more complexity on its own with plugin configuration because of Spring.

May be it should be split up to two tasks: one for fragments with plain Web project and another task for Spring.

---

@pleku commented on [Fri Jan 12 2018](https://github.com/vaadin/flow-maven-plugin/issues/41#issuecomment-357236577)

We'll need to get the plugin to work in Bakery anyway, so let's fix the plugin for that.

---

@denis-anisimov commented on [Fri Jan 12 2018](https://github.com/vaadin/flow-maven-plugin/issues/41#issuecomment-357315414)

Sure.
Then as the first task the plugin should be fixed so that it works with Bakery and then Bakery should be updated to use the pugin.
After that this task can be taken again.
So the task is blocked by:
- fix maven plugin so that it works with bakery
- update bakery to use the plugin

**Comment by SomeoneToIgnore** [comment]
One of the three tickets mentioned in the original issue is closed (by moving flow-maven-plugin into flow repo), the other two tickets were moved from the plugin repo to flow repo and are open now:
https://github.com/vaadin/flow/issues/3321
https://github.com/vaadin/flow/issues/3315

**Comment by pleku** [comment]
### Acceptance Criteria
There is a PR for Bakery that adds the flow-maven-plugin with bundling working.

**Comment by pekam** [comment]
The bundling is working in this PR: https://github.com/vaadin/bakery-app-starter-flow-spring/pull/398
But the fragments are not, as explained in https://github.com/vaadin/flow/issues/3457

**Comment by pleku** [comment]
Two issues remaining for bundling:
https://github.com/vaadin/flow/issues/3457
https://github.com/vaadin/flow/issues/3453
